com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollServiceInProgressOperationsExecution.java

```diff
@@ -7,13 +7,15 @@
 
 import org.cloudfoundry.client.lib.CloudControllerClient;
 import org.cloudfoundry.client.lib.domain.CloudEvent;
+import org.flowable.engine.delegate.DelegateExecution;
 
 import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceExtended;
 import com.sap.cloud.lm.sl.cf.core.cf.clients.EventsGetter;
 import com.sap.cloud.lm.sl.cf.core.cf.clients.ServiceGetter;
 import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperation;
 import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationState;
 import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationType;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
 import com.sap.cloud.lm.sl.cf.process.util.ServiceOperationExecutor;
 
 public class PollServiceInProgressOperationsExecution extends PollServiceOperationsExecution implements AsyncExecution {
```

Comment:
Lets move the logic from this class to its parent and rename it back to Poll delete.....
That way we are going to have a generic way of fetching lastOperations.
Then the responsibilities of the two subclasses (poll delete and poll create/update) will be to handle the evaluation of the lastOperation (Success/Fail/Complete) and handle the states (Decide if the polling should continue / throw exception / complete silently).
Think about it. An UML or some drawing somewhere of all of the classes involved and the responsibilities is going to help a lot !